### PR TITLE
Add D version of High_IQ (48).

### PR DESCRIPTION
**Pull request by veelo**

### CODE REVIEW

**Comment by coding-horror** [comment]
a special thanks for the extensive `readme.md` notes! 🙏

**Comment by veelo** [comment]
Thank you :-)

File: 48_High_IQ/d/highiq.d [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    return readln;
+}
+
+version (Windows)
+{
+    // Make the Windows console do a better job at printing UTF-8 strings,
+    // and restore the default upon termination.
+
+    import core.sys.windows.windows;
+
```
Comment (K780420) [inline 48_High_IQ/d/highiq.d:228]: image

File: 48_High_IQ/d/highiq.d [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+{
+    return readln;
+}
+
+version (Windows)
+{
+    // Make the Windows console do a better job at printing UTF-8 strings,
+    // and restore the default upon termination.
+
+    import core.sys.windows.windows;
```
Comment (K780420) [inline 48_High_IQ/d/highiq.d:227]: image

**Review by K780420 — COMMENTED** [review]
image

File: 48_High_IQ/d/highiq.d [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    shared static this() @trusted
+    {
+        SetConsoleOutputCP(CP_UTF8);
+    }
+
+    shared static ~this() @trusted
+    {
+        SetConsoleOutputCP(GetACP);
+    }
+}
```
Comment (K780420) [inline 48_High_IQ/d/highiq.d:238]: image

**Review by K780420 — COMMENTED** [review]
image

**Review by K780420 — COMMENTED** [review]
image

**Pull request merged**